### PR TITLE
Bump base docker image to 0.6.0.

### PR DESCRIPTION
**Pull request by yakutovicha**
This is a cherry-pick of (#5609)

The new image is built on top of Ubuntu 22.04. It comes with PostgreSQL 10 and
RabbitMQ 3.8.14 to facilitate the migration to AiiDA 2.0. It supports the `amd64`
and `arm64` architectures.

### CODE REVIEW

**Comment by yakutovicha** [comment]
Once this PR is merged, it would be good if AiiDA 2.0.3 could be released. We will base the new AiiDAlab docker image on top of it.

**Review by sphuber — APPROVED** [review]

**Comment by sphuber** [comment]
@eimrek do you have the time to make the v2.0.3 release? I finished updating and cleaning the release wiki](https://github.com/aiidateam/aiida-core/wiki/How-to-make-a-new-release) so you would be good to go over it to see if it is correct/clear/complete.

@yakutovicha any other commits that should go in v2.0.3 or this is the only one as far as you are concerned?

**Comment by eimrek** [comment]
> @eimrek do you have the time to make the v2.0.3 release? I finished updating and cleaning [the release wiki](https://github.com/aiidateam/aiida-core/wiki/How-to-make-a-new-release) so you would be good to go over it to see if it is correct/clear/complete.

Sure, will do after Sasha replies.

**Comment by yakutovicha** [comment]
> @yakutovicha any other commits that should go in v2.0.3 or this is the only one as far as you are concerned?

Only this change was needed for us 👍

**Pull request merged**
